craft_parts/plugins/python_plugin.py

```diff
@@ -153,7 +153,9 @@ def get_build_commands(self) -> List[str]:
             requirements_cmd = f"{pip} install {constraints} -U {requirements}"
             build_commands.append(requirements_cmd)
 
-        build_commands.append(f"[ -f setup.py ] && {pip} install {constraints} -U .")
+        build_commands.append(
+            f"[ -f setup.py ] || [ -f pyproject.toml ] && {pip} install {constraints} -U ."
```

Comment:
```suggestion
            f"[ -f setup.py -o -f pyproject.toml ] && {pip} install {constraints} -U ."
```

This is a bit cleaner IMO and allows an easier update to include `setup.cfg` files if we choose.